### PR TITLE
ci: trigger mender-conductor-enterprise travis job

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Signed-off-by: Gregorio Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

**Review by kjaskiewiczz — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -107,3 +107,9 @@ jobs:
               on:
                   tags: true
                   all_branches: true
+
+      after_deploy:
```
Comment (maciejmrowiec) [inline .travis.yml]: Not that build stage, it should be part of "DOCKER_REPOSITORY=mendersoftware/mender-conductor" build

Comment (GregorioDiStefano) [reply .travis.yml]: good catch, fixed!

File: trigger-mender-conductor-enterprise.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -u -x -e
+
+body='{
+"request": {
+"branch":'"$TRAVIS_BRANCH"'
```
Comment (maciejmrowiec) [inline trigger-mender-conductor-enterprise.sh]: add env with mender-conductor revision (eg `COMMIT` variable content)
Outside of branch / tag builds, master in mender-conductor-enterprise needs to produce: `master_<mender-conductor-sha>_ <mender-contuctor-enterprise-sha>` to have unique tags for master for the combination of those two

**Comment by maciejmrowiec** [comment]
btw that looks surprisingly easy to trigger ;)

File: .travis.yml [hunk 3]
```diff
@@ -68,6 +68,13 @@ jobs:
                   tags: true
                   all_branches: true
 
+      after_deploy:
+          -  export TRAVIS_BRANCH
```
Comment (maciejmrowiec) [inline .travis.yml]: use IMAGE_TAG var, that will have tags and branches
var should be already exported to env in line 43

File: trigger-mender-conductor-enterprise.sh (hunk 2 above)
Comment (maciejmrowiec) [reply trigger-mender-conductor-enterprise.sh]: You can set arbitrary envs accrdign to docs:

```
body='{
 "request": {
 "message": "Override the commit message: this is an api request",
 "branch":"master",
 "config": {
   "env": {
     "matrix": ["TEST=unit"]
   },
   "script": "echo FOO"
  }
}}'

{
  "env": {
    "BAZ": "baz_from_api_request"
  },
  "script": "echo FOO"
}
```

Comment (maciejmrowiec) [reply trigger-mender-conductor-enterprise.sh]: https://docs.travis-ci.com/user/triggering-builds/

**Review by mchalski — COMMENTED** [review]
since @maciejmrowiec is not here I'll try to do the review @GregorioDiStefano 

if I understand correctly:
- IMAGE_TAG env is picked from this repo
- it is passed via the api to the `-enterprise` job
- ...which recognizes it and works it into the build

it seems almost ok, but this howto:  https://docs.travis-ci.com/user/triggering-builds/

tells me that env vars should be passed like:
```
"config": {
      "env": {
        "BAZ": "baz_from_api_request"
      }
``` 

did you test this stuff, i.e. verified that a triggered enterprise image indeed has *this* image as base?

if it does work, then disregard my comment

**Comment by GregorioDiStefano** [comment]
@mchalski yup, figured that bit out  (changed locally), but, since we are using a multi-stage build, only the first build is getting the environment variable set (!!), and the "Build and Push" stage doesn't have the variable.

**Pull request merged**